### PR TITLE
Fix document resolution in DocumentRouteHandler

### LINKED ISSUE #16420: [Bug]: Having a pretty URL, that matches the key of a document link leads to infinite redirects on multisite environments
### Pimcore version

11.1.4

### Steps to reproduce

Create a subsite. Create a link document with the key "test". Create a directory or a page and nest another page with the key "test" below it. Give the nested "test" document the pretty URL "/test".

### Actual Behavior

I am caught in an infinite redirect loop.

### Expected Behavior

Navigating to /test should show the page with the internal path /subsite/subfolderOrPage/test.

### PR DESCRIPTION
**Pull request by alexej-d**
## Changes in this pull request  
Resolves #16420

### CODE REVIEW

File: lib/Routing/Dynamic/DocumentRouteHandler.php [hunk 1]
```diff
@@ -102,7 +102,7 @@ public function getRouteByName(string $name): ?DocumentRoute
 
     public function matchRequest(RouteCollection $collection, DynamicRequestContext $context): void
     {
-        $document = Document::getByPath($context->getPath());
+        $document = Document\Page::getByPath($context->getPath());
```
Comment (mattamon) [inline lib/Routing/Dynamic/DocumentRouteHandler.php]: Is it possible to import the Page as a whole instead of Document\Page?

**Comment by mattamon** [comment]
@alexej-d could you please merge the latest changes from 11.3 into your branch to resolve all the conflicts? :)

**Comment by alexej-d** [comment]
> @alexej-d could you please merge the latest changes from 11.3 into your branch to resolve all the conflicts? :)

All done :)

**Comment by brusch** [comment]
Closing as obsolete, see https://github.com/pimcore/pimcore/issues/16420#issuecomment-2454130256 
@alexej-d thanks anyway for your efforts 🙏

**Pull request closed without merging**